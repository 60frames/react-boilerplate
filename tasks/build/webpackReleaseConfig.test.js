/* eslint-env jest */

const webpackReleaseConfig = require('./webpackReleaseConfig');
const webpackConfigSnapshotSerializer = require('../../test/webpackConfigSnapshotSerializer');

describe('tasks/build/webpackReleaseConfig', () => {
  expect.addSnapshotSerializer(webpackConfigSnapshotSerializer);

  describe('client config', () => {
    it('has not regressed', () => {
      const config = webpackReleaseConfig.find(
        config => config.name === 'client'
      );
      expect(config).toBeDefined();
      expect({
        __webpack_config__: config
      }).toMatchSnapshot();
    });
  });

  describe('server config', () => {
    it('has not regressed', () => {
      const config = webpackReleaseConfig.find(
        config => config.name === 'server'
      );
      expect(config).toBeDefined();
      delete config.externals; // Reduce noise from node_modules
      expect({
        __webpack_config__: config
      }).toMatchSnapshot();
    });
  });
});
