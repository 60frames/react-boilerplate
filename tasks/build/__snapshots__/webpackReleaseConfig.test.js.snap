// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tasks/build/webpackReleaseConfig client config has not regressed 1`] = `
Object {
  "bail": true,
  "context": "<rootDir>/src",
  "devtool": "",
  "entry": Object {
    "client": "./client",
  },
  "externals": Object {},
  "module": Object {
    "rules": Array [
      Object {
        "include": "<rootDir>/src",
        "loader": "babel-loader",
        "options": Object {
          "cacheDirectory": true,
        },
        "test": /\\\\\\.js\\$/,
      },
      Object {
        "include": /node_modules/,
        "test": /\\\\\\.css\\$/,
        "use": Array [
          Object {
            "loader": "<rootDir>/node_modules/extract-text-webpack-plugin/loader.js",
            "options": Object {
              "omit": 1,
              "remove": true,
            },
          },
          Object {
            "loader": "style-loader",
          },
          Object {
            "loader": "css-loader",
            "options": Object {
              "localIdentName": "[hash:base64]",
              "minimize": true,
              "modules": false,
              "sourceMap": false,
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "ident": "postcss",
              "plugins": [Function],
            },
          },
        ],
      },
      Object {
        "include": "<rootDir>/src",
        "test": /\\\\\\.css\\$/,
        "use": Array [
          Object {
            "loader": "<rootDir>/node_modules/extract-text-webpack-plugin/loader.js",
            "options": Object {
              "omit": 1,
              "remove": true,
            },
          },
          Object {
            "loader": "style-loader",
          },
          Object {
            "loader": "css-loader",
            "options": Object {
              "localIdentName": "[hash:base64]",
              "minimize": true,
              "modules": true,
              "sourceMap": false,
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "ident": "postcss",
              "plugins": [Function],
            },
          },
        ],
      },
      Object {
        "include": "<rootDir>/src",
        "loader": "url-loader",
        "options": Object {
          "limit": 10000,
        },
        "test": /\\\\\\.\\(eot\\|ttf\\|woff\\|woff2\\)\\(\\\\\\?\\.\\*\\)\\?\\$/,
      },
      Object {
        "include": "<rootDir>/src",
        "loader": "url-loader",
        "options": Object {
          "limit": 10000,
        },
        "test": /\\\\\\.\\(gif\\|ico\\|jpg\\|jpeg\\|png\\|svg\\|webp\\)\\$/,
      },
      Object {
        "include": "<rootDir>/src",
        "loader": "url-loader",
        "options": Object {
          "limit": 10000,
        },
        "test": /\\\\\\.\\(mp4\\|webm\\)\\$/,
      },
      Object {
        "include": "<rootDir>/src",
        "loader": "url-loader",
        "options": Object {
          "limit": 10000,
        },
        "test": /\\\\\\.\\(aac\\|m4a\\|mp3\\|oga\\|ogg\\|wav\\)\\$/,
      },
    ],
  },
  "name": "client",
  "node": Object {
    "process": false,
  },
  "output": Object {
    "filename": "[name].[chunkhash].js",
    "libraryTarget": "var",
    "path": "<rootDir>/dist",
    "publicPath": "",
  },
  "plugins": Array [
    ExtractTextPlugin {
      "filename": "[name].[contenthash].css",
      "id": 1,
      "options": Object {
        "allChunks": true,
        "disable": false,
      },
    },
    UglifyJsPlugin {
      "options": Object {
        "compress": Object {
          "screw_ie8": true,
          "warnings": false,
        },
        "mangle": Object {
          "screw_ie8": true,
        },
        "output": Object {
          "comments": false,
          "screw_ie8": true,
        },
        "sourceMap": false,
      },
    },
    DefinePlugin {
      "definitions": Object {
        "process.env.NODE_ENV": "\\"production\\"",
      },
    },
    HashedModuleIdsPlugin {
      "options": Object {
        "hashDigest": "base64",
        "hashDigestLength": 4,
        "hashFunction": "md5",
      },
    },
    StatsPlugin {
      "cache": Object {},
      "options": Object {
        "chunkModules": true,
      },
      "output": "client-stats.json",
    },
    CommonsChunkPlugin {
      "async": undefined,
      "children": undefined,
      "chunkNames": Array [
        "bootstrap",
      ],
      "filenameTemplate": "[name].[chunkhash].js",
      "ident": "<rootDir>/node_modules/webpack/lib/optimize/CommonsChunkPlugin.js0",
      "minChunks": Infinity,
      "minSize": undefined,
      "selectedChunks": undefined,
    },
  ],
  "resolve": Object {
    "modules": Array [
      "<rootDir>/src",
      "node_modules",
    ],
  },
  "target": "web",
}
`;

exports[`tasks/build/webpackReleaseConfig server config has not regressed 1`] = `
Object {
  "bail": true,
  "context": "<rootDir>/src",
  "devtool": "",
  "entry": Object {
    "server": "./server",
  },
  "module": Object {
    "rules": Array [
      Object {
        "include": "<rootDir>/src",
        "loader": "babel-loader",
        "options": Object {
          "cacheDirectory": true,
        },
        "test": /\\\\\\.js\\$/,
      },
      Object {
        "include": /node_modules/,
        "test": /\\\\\\.css\\$/,
        "use": Array [
          Object {
            "loader": "css-loader/locals",
            "options": Object {
              "localIdentName": "[hash:base64]",
              "minimize": true,
              "modules": false,
              "sourceMap": false,
            },
          },
        ],
      },
      Object {
        "include": "<rootDir>/src",
        "test": /\\\\\\.css\\$/,
        "use": Array [
          Object {
            "loader": "css-loader/locals",
            "options": Object {
              "localIdentName": "[hash:base64]",
              "minimize": true,
              "modules": true,
              "sourceMap": false,
            },
          },
        ],
      },
      Object {
        "include": "<rootDir>/src",
        "loader": "url-loader",
        "options": Object {
          "limit": 10000,
        },
        "test": /\\\\\\.\\(eot\\|ttf\\|woff\\|woff2\\)\\(\\\\\\?\\.\\*\\)\\?\\$/,
      },
      Object {
        "include": "<rootDir>/src",
        "loader": "url-loader",
        "options": Object {
          "limit": 10000,
        },
        "test": /\\\\\\.\\(gif\\|ico\\|jpg\\|jpeg\\|png\\|svg\\|webp\\)\\$/,
      },
      Object {
        "include": "<rootDir>/src",
        "loader": "url-loader",
        "options": Object {
          "limit": 10000,
        },
        "test": /\\\\\\.\\(mp4\\|webm\\)\\$/,
      },
      Object {
        "include": "<rootDir>/src",
        "loader": "url-loader",
        "options": Object {
          "limit": 10000,
        },
        "test": /\\\\\\.\\(aac\\|m4a\\|mp3\\|oga\\|ogg\\|wav\\)\\$/,
      },
    ],
  },
  "name": "server",
  "node": Object {
    "process": false,
  },
  "output": Object {
    "filename": "[name].js",
    "libraryTarget": "commonjs2",
    "path": "<rootDir>/dist",
    "publicPath": "",
  },
  "plugins": Array [
    ExtractTextPlugin {
      "filename": "[name].css",
      "id": 2,
      "options": Object {
        "allChunks": true,
        "disable": true,
      },
    },
    UglifyJsPlugin {
      "options": Object {
        "compress": Object {
          "screw_ie8": true,
          "warnings": false,
        },
        "mangle": Object {
          "screw_ie8": true,
        },
        "output": Object {
          "comments": false,
          "screw_ie8": true,
        },
        "sourceMap": false,
      },
    },
    DefinePlugin {
      "definitions": Object {
        "process.env.NODE_ENV": "\\"production\\"",
      },
    },
    HashedModuleIdsPlugin {
      "options": Object {
        "hashDigest": "base64",
        "hashDigestLength": 4,
        "hashFunction": "md5",
      },
    },
    LimitChunkCountPlugin {
      "options": Object {
        "maxChunks": 1,
      },
    },
  ],
  "resolve": Object {
    "modules": Array [
      "<rootDir>/src",
      "node_modules",
    ],
  },
  "target": "node",
}
`;
